### ISSUE TITLE
Module for classifiers and interaction analysis

### ISSUE THREAD

**Issue by adiah80** [SforAiDl/twitter-sanity#6]
Building upon issues #4 and #5, we need a general module that explores several classification models. 

Some libraries that can be explored for classification models are [scikit-learn](https://scikit-learn.org/stable/index.html), [xgboost](https://xgboost.readthedocs.io/en/latest/), and [catboost](https://catboost.ai/) among others. 

Other NLP based methods can also be explored for analyzing the user's preferences. These would employ the tweet text and user interaction metrics.

**Comment by shashank-m** [comment]
I can take this issue up. Can someone tell me what the inputs and outputs of the model are?

**Comment by ajaysub110** [comment]
Inputs to the model are the tweet text, username of tweeter, number of existing retweets and likes. Output label is binary (1 or 0) determining whether the tweet is interesting or not to our user. These labels have been assigned in the train set based on whether has interacted (liked or retweeted) a tweet or not.

**Comment by ajaysub110** [comment]
Of course, we could do generate some of our own features from these too and/or introduce more inputs (if available) to the model. That's up to you

**Comment by ajaysub110** [comment]
@adiah80 had suggested collaborative ranking/clustering which is used in recommendation systems. This paper will be probably be useful to check out first
https://homes.cs.washington.edu/~tqchen/data/pdf/sigir12-p661-chen.pdf

**Comment by shashank-m** [comment]
Yeah cool. But I was thinking model output will be topic interested in. Like a multiclass classification problem

**Comment by ajaysub110** [comment]
What do you think can be good outputs for our model?

**Comment by shashank-m** [comment]
Hmm but if the output is topic then labelling will be hard i guess

**Comment by ajaysub110** [comment]
Oh you meant "topic interested in" as an output? Yeah, only trends are assigned topics on twitter, I think

**Comment by shashank-m** [comment]
But I think topics can be a feature in the train set. Like we can cluster the tweets in the train set and assign cluster number of the tweet as a feature. How does this sound?

**Comment by ajaysub110** [comment]
Hmm yeah we could try that
